tradesman/model_creation/import_network.py

```diff
@@ -1,31 +1,185 @@
+import gc
+import csv
+from tempfile import gettempdir
+from time import sleep
+import osm2gmns as og
 from aequilibrae import Project, Parameters
+from os.path import join
+
+import pandas as pd
+import requests
 
-from tradesman.data_retrieval.country_main_area import country_border_from_model
 from tradesman.model_creation.extra_data_fields import extra_fields
+from tradesman.data_retrieval.osm_tags.set_bounding_boxes import bounding_boxes
+
+
+class ImportNetwork:
+    def __init__(self, project: Project, model_place: str, pbf_path: str = None):
+
+        self.project = project
+        self.model_place = model_place
+        self.pbf_path = pbf_path
+        self.json = []
+        self.par = Parameters()
+        self.new_link_fields = {
+            "osm_way_id": {"description": "osm_id", "type": "text", "required": False},
+        }
+        self.new_node_fields = {"osm_node_id": {"description": "osm_id", "type": "text", "required": False}}
+
+    def import_network(self):
+        """
+        Imports network from OSM or allows the user to manually import network from .osm or .pbf files.
+
+        Args.:
+            *project*(:obj:`aequilibrae.project.Project`):
+            *model_place*(:obj:`str`):
+            *pbf_path*(:obj:`str`): path to osm or pbf file. Optional.
+
+        """
+
+        if self.pbf_path:
+            print("Convert to GMNS ...")
+            print(" ")
+            net = og.getNetFromFile(self.pbf_path, network_types=("auto"))  # Fix it later
+            og.outputNetToCSV(net, output_folder=gettempdir(), prefix=f"{self.model_place}-gmns-", encoding="utf-8")
+
+            print(" ")
+            print("Adjust GMNS files ...")
+            print(" ")
+            self.__adjust_link_file(
+                file_path=join(gettempdir(), f"{self.model_place}-gmns-link.csv"), model_place=self.model_place
+            )
+
+            self.par.parameters["network"]["gmns"]["link"]["fields"].update(self.new_link_fields)
+            self.par.parameters["network"]["gmns"]["node"]["fields"].update(self.new_node_fields)
+            self.par.write_back()
+
+            print("Create network from GMNS ...")
+            print(" ")
+            self.project.network.create_from_gmns(
+                link_file_path=join(gettempdir(), f"{self.model_place}-gmns-link.csv"),
+                node_file_path=join(gettempdir(), f"{self.model_place}-gmns-node.csv"),
+            )
+
+            print(" ")
+            print("Download OSM data for bridges, tolls and tunnels ...")
+            print(" ")
+            self.__download_osm_data(model_place=self.model_place)
+
+            print("Add new columns ...")
+            print(" ")
+            self.__setup_tables()
+
+            print("Update links ...")
+            print(" ")
+            self.__update_links()
+
+        else:
```

Comment:
This is something that can be simpler with an "early exit".

    if not self.pbf_path:
        self.par.parameters["network"]["links"]["fields"]["one-way"].extend(extra_fields)
        self.par.write_back()
        self.project.network.create_from_osm(place_name=self.model_place)
        return

And then the rest of the code does not need to be wrapped inside an *if* statement